src/transaction.js

```diff
@@ -45,6 +46,13 @@ class Transaction {
         else {
           note = new Uint8Array(0);
         }
+        if (lease !== undefined) {
+            if (lease.constructor !== Uint8Array) throw Error("lease must be a Uint8Array.");
+            if (lease.length !== ALGORAND_TRANSACTION_LEASE_LENGTH) throw Error("lease must be of length " + ALGORAND_TRANSACTION_LEASE_LENGTH.toString() + ".");
+        }
+        else {
+            lease = new Uint8Array(0);
+        }
```

Comment:
It's a duplicate of the `addLease` function. remove it and keep just the other one.

Reply:
@rotem I'm removing it from the helpers, but would like to keep it as an optional constructor arg like this.

A common way of building and signing transactions is building constructor arguments in a dict and then passing the constructors to `algosdk.signTransaction` to immediately get the signed transaction back - see example in the test `it('should return a blob that matches the go code when using a flat fee'`.

`AddLease` can't be called on a signed transaction, though, and I wanted to support the `algosdk.signTransaction` path.

Reply:
I'm ok with the constructor but the one thing we can't do is to break the API. This SDK update should be backwards compatible so we can't change exiting ones. One unfortunate result is that we will not be able to support the `algosdk.signTransaction` path. 

Bottom line - We can leave the constructor trick but we can't change any of the `main.js` functions.

Reply:
Sounds good - current commit keeps the constructor trick but leaves `main.js` as is